src/Symfony/Component/Cache/Adapter/RedisTagAwareAdapter.php

```diff
@@ -305,4 +306,185 @@ private function getRedisEvictionPolicy(): string
 
         return $this->redisEvictionPolicy = '';
     }
+
+    private function getPrefix(): string
+    {
+        if ($this->redis instanceof \Predis\ClientInterface) {
+            $prefix = $this->redis->getOptions()->prefix ? $this->redis->getOptions()->prefix->getPrefix() : '';
+        } elseif (\is_array($prefix = $this->redis->getOption(\Redis::OPT_PREFIX) ?? '')) {
+            $prefix = current($prefix);
+        }
+
+        return $prefix;
+    }
+
+    /**
+     * Returns all existing tag keys from the cache.
+     *
+     * @TODO Verify the LUA scripts are redis-cluster safe.
+     */
+    protected function getAllTagKeys(): array
+    {
+        $tagKeys = [];
+        $prefix = $this->getPrefix();
+        // need to trim the \0 for lua script
+        $tagsPrefix = trim(self::TAGS_PREFIX);
+
+        // get all SET entries which are tagged
+        $getTagsLua = <<<'EOLUA'
+            redis.replicate_commands()
+            local cursor = ARGV[1]
+            local prefix = ARGV[2]
+            local tagPrefix = string.gsub(KEYS[1], prefix, "")
+            return redis.call('SCAN', cursor, 'COUNT', 5000, 'MATCH', '*' .. tagPrefix .. '*', 'TYPE', 'set')
+        EOLUA;
+        $cursor = null;
```

Comment:
This should be 0, null leads to something like: `ERR invalid cursor script: 5880028e0d564320af1c1aabb595c9985f5a1a04, on @user_script:5.`
This is used in SCAN which expects 0 to start and returns 0 when the iteration is finished: https://redis.io/docs/latest/commands/scan/ 
Unfortunately I can't find older docs and have to assume this is backward compatible.